File: 2021/May/May 8/unique-occurences.py
# Source : https://leetcode.com/problems/unique-number-of-occurrences/submissions/
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occurences = [arr.count(i) for i in set(arr)]
        return len(occurences)==len(set(occurences))

File: 2022/November/Day 30/uniqueOccurences.cpp
// Source : https://leetcode.com/problems/unique-number-of-occurrences/submissions/
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {

        map<int,int> m1;
        map<int,int> m2;
        for(auto&i:arr)
        {
            m1[i]+=1;
        }

        for (const auto &[key, value]: m1 ) {
       if(m2[value]!=0)
       {
           return false;
       }
            else
            {
                m2[value]=key;
            }
    }
        return true;

    }
};
